yapapi/services.py

```diff
@@ -592,7 +593,12 @@ def _get_handler(instance: ServiceInstance):
         }
         handler = _handlers.get(instance.state, None)
         if handler:
-            return handler()
+            if inspect.isasyncgenfunction(handler):
+                return handler()
+            else:
+                raise ServiceError(
+                    f"Service handler: `{handler}` must be an asynchronous generator."
+                )
```

Comment:
```suggestion
                service_cls_name = type(instance.service).__name__
                handler_name = handler.__name__
                raise ServiceError(
                    f"Service handler: `{service_cls_name}.{handler_name}` must be an asynchronous generator."
                )

```

Also I would consider rising `TypeError` instead of `ServiceError`.